src/mds/objects.py

```diff
@@ -391,43 +392,63 @@ async def delete_object(
         request (Request): starlette request (which contains reference to FastAPI app)
     Returns:
         204: if record and metadata are deleted
-        403: if fence returns a 403 unauthorized response
-        500: if fence does not return 204 or 403 or there is an error deleting metadata
+        403: if fence/indexd returns a 403 unauthorized response
+        500: if fence/indexd does not return 204 or 403 or there is an error deleting metadata
     """
+    delete_file_locations = False
+    if "delete_file_locations" in request.query_params:
+        if request.query_params["delete_file_locations"]:
+            raise HTTPException(
+            HTTP_400_BAD_REQUEST, f"Query param `delete_file_locations` should not contain any value"
+        )
+        delete_file_locations = True
+    svc_name = "fence" if "delete_file_locations" in request.query_params else "indexd"
     try:
-        fence_endpoint = urljoin(config.DATA_ACCESS_SERVICE_ENDPOINT, f"data/{guid}")
         auth_header = str(request.headers.get("Authorization", ""))
         headers = {"Authorization": auth_header}
-        fence_response = await request.app.async_client.delete(
-            fence_endpoint, headers=headers
-        )
-        fence_response.raise_for_status()
+        if delete_file_locations:
+            fence_endpoint = urljoin(config.DATA_ACCESS_SERVICE_ENDPOINT, f"data/{guid}")
+            response = await request.app.async_client.delete(
+                fence_endpoint, headers=headers
+            )  
+        else:
+            rev = await get_indexd_revision(guid, request)
+            indexd_endpoint = urljoin(config.INDEXING_SERVICE_ENDPOINT, f"index/{guid}")
+            response = await request.app.async_client.delete(
+                indexd_endpoint, params = {"rev": rev}, headers=headers
+            )
+        response.raise_for_status()
         await (
-            Metadata.delete.where(Metadata.guid == guid)
-            .returning(*Metadata)
-            .gino.first()
+        Metadata.delete.where(Metadata.guid == guid)
+        .returning(*Metadata)
+        .gino.first()
         )
         return JSONResponse({}, HTTP_204_NO_CONTENT)
     except httpx.HTTPError as err:
         logger.debug(err)
         if err.response:
             raise HTTPException(
-                err.response.status_code, {"fence_response": err.response.text}
+                err.response.status_code, {f"{svc_name}_response": err.response.text}
             )
         raise HTTPException(
-            HTTP_500_INTERNAL_SERVER_ERROR, "Error during request to fence"
+            HTTP_500_INTERNAL_SERVER_ERROR, f"Error during request to {svc_name}"
         )
 
+async def get_indexd_revision(guid, request):
+    response = await get_object(guid, request)
+    indexd_document =  json.loads(response.body)
+    return indexd_document['record']['rev'] if "record" in indexd_document and "rev" in indexd_document['record'] else None
```

Comment:
quick note that `return indexd_document.get('record', {}).get('rev')` would work

But i think you should call indexd directly here (like [this](https://github.com/uc-cdis/metadata-service/blob/afa646d796ff4d9b83b12af6cb64ce080d9ea9bb/src/mds/objects.py#L361-L363)), instead of going through `get_object()` which does more than what you need (such as querying the MDS DB) which impacts complexity and performance